openqaoa/workflows/optimizer.py

```diff
@@ -56,6 +59,18 @@ class Optimizer(ABC):
             A list containing the available cloud providers
         compiled: `Bool`
             A boolean flag to check whether the optimizer object has been correctly compiled at least once     
+        id: `dict`
+            A dictionary containing the identification stamps of the optimizer object. 
+            'uuid' is the unique identifier of the optimizer object, 
+            'parent_uuid' is the unique identifier of the parent object (e.g. the project to which the optimizer object belongs), the user can set this value using the set_identification method, 
```

Comment:
Maybe `experiment_uuid`? It will be more explicit that way?

Reply:
I used `parent_uuid`, because if we are running a RQAOA the intermediate QAOA will have as a `parent_uuid` the uuid of the parent RQAOA.
But we can change the approach if you want.

Reply:
We need both: we need a run_id, and an experiment_id.
- run_id is the id of the experiment. If qaoa, then run_id is job id of the qaoa you want to run; if rqaoa then the run_id is the overall rqaoa id, and each qaoa will have its own set of ids
- exp_id is the overall id for the experiment

I am happy to discuss finding better names for these two ids though!

Reply:
I used "parent_uuid" for both types of uuid because I was imagining a table in the database like this one:
https://www.notion.so/entropicawiki/OQ-database-e678ba6dd4d644eab40da2af00bcf0ed#b05158dcc5554bf5b5d2de85ef474728
With that you would be able to see the hierarchy.


However, depending on how we dump the data to the database all of this discussion is not useful. This is because, if we dump into the database at each RQAOA step then we need the id of the RQAOA in each QAOA. But, if we dump after the whole RQAOA, then we don't need the QAOA to contain any uuid because it will be in the RQAOA info.

Reply:
The atomic experiment in openqaoa will be a QAOA. That is, all the measurement records for however many optimisations are needed to terminate the optimisation.

So, if RQAOA has 10 limitations we will save 10 files. Each QAOA will also need to reference the correct RQAOA id.

Reply:
ok, but for now this is not implemented this way, because you can dump only when RQAOA is done, so we should rethink how we do the dumping, no?

Regarding the ids: would it make sense to have, "experiment_uuid" (the id of this workflow), "project_uuid" (the project id), and "parent_uuid" (this would be empty for RQAOA and single QAOA, but filled with "experiment_uuid" for QAOA steps in RQAOAs)